puppet/lib/puppet/indirector/facts/puppetdb.rb

```diff
@@ -26,7 +26,15 @@ def save(request)
         facts = request.instance.dup
         facts.values = facts.values.dup
         facts.values[:trusted] = get_trusted_info(request.node)
-        {
+
+        inventory = facts.values['_puppet_inventory_1']
+        package_inventory = nil
+        if inventory
```

Comment:
if _puppet_inventory_1 is null, we still want to delete it I think. Does this guard against that?